src/core/common/uptime.cpp

```diff
@@ -107,20 +107,20 @@ static uint32_t DivideAndGetRemainder(uint32_t &aDividend, uint32_t aDivisor)
 
     aDividend -= quotient * aDivisor;
 
-    return quotient;
+    return static_cast<uint16_t>(quotient);
 }
 
 void Uptime::UptimeToString(uint64_t aUptime, StringWriter &aWriter)
 {
     uint64_t days = aUptime / Time::kOneDayInMsec;
     uint32_t remainder;
-    uint32_t hours;
-    uint32_t minutes;
-    uint32_t seconds;
+    uint16_t hours;
+    uint16_t minutes;
+    uint16_t seconds;
 
     if (days > 0)
     {
-        aWriter.Append("%lud.", days);
+        aWriter.Append("%llud.", static_cast<unsigned long long>(days));
```

Comment:
`long long` type is pretty tricky to use, its size depends on platforms and compiler options, it can lead to some strange when the libc is compiled with different compiler options against the program.

So it is suggested to use `PRIxxx` macros rather than `long long` (`PRIu64` here and no type cast). Their wide is determined from its name, without ambitious or confusion.

Reply:
Thanks. Yes. I was looking at the `PRIxxx` model as well. I know that can also be tricky to use (it is not always available on all toolchains and library header). 

One idea may be to avoid printing `uint64` values when/if we can or convert it to smaller types before printing.

For example here, we can cast and limit the `days` to` uint32_t`  (`2^32` days is already more than 11 million years 😄).

I know we did this in other place `string.Append("0x%x-%08x", static_cast<uint32_t>(mValue >> 32), static_cast<uint32_t>(mValue));` (which make the printed output easier to read as well).

I will change the PR in next push to use this model (we still need to cast `uint32_t` to `unsigned long` when we print it but I think that should be fine).

Reply:
Yes, PRIu64 is also tricky, such that matter code forbids its usage, and declare several macros to handle it.

https://github.com/project-chip/connectedhomeip/blob/0a0865130c80b5772727a5b77c7f265ed6c07715/src/lib/support/logging/CHIPLogging.h#L306-L339